src/pages/Account/Profile.tsx

```diff
@@ -44,8 +45,10 @@ const Profile: React.FunctionComponent<PropsType> = ({ location }) => {
   const onSubmit = async (data: ProfileFormData) => {
     if (!user) return;
     await updateUserProfile(user.uid, data);
-    const accountQuestionsUrl = `/account/questions?venueId=${venueId}&returnUrl=${window.location.pathname}${window.location.search}`;
-    const nextUrl = venueId ? accountQuestionsUrl : `/${DEFAULT_VENUE}`;
+    const accountQuestionsUrl = `/account/questions?venueId=${venueId}${
+      returnUrl ? "&returnUrl=" + returnUrl : ""
+    }`;
```

Comment:
Bringing `returnUrl` along through the flow here.

Reply:
[Future] We can use https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams and similar to handle this in a better way than string concatenation.